### DISCUSSION TITLE
Crash on macOS if you move Ghostty.app while Ghostty is running

### DISCUSSION THREAD

**Discussion by jellevandenhooff** [ghostty-org/ghostty#3007] [Feature Requests, Ideas] [answered]
After you move the Ghostty.app folder while the application is running, opening a new window crashes the application. When you press command-N to open a new window, all terminals freeze and nothing except a force quit makes progress. I don't know how actionable this crash is, so I made this a discussion instead of an issue. Cursory Googling does not immediately mention solutions.

I experienced this crash after I downloaded Ghostty.app and accidentally installed (moved) it into a subfolder of Applications. I noticed the path, moved the binary, and then triggered the above crash. MacOS warns if you rename an application while it is running, but it does not warn when you move an application.

**Comment by taylrfnt** [comment]
This is reproducible (both `tip` and `stable`), but frankly I am not exactly sure what the use case is here.  I would assume that the vast majority of users are not moving their `Ghostty.app` while it is running, and even if they did experience this crash, the app re-opens just fine from its new location.

Anyway, here's a crash report dump for this behavior on a local build I reproduced this with (I had to zip so GitHub accepts it as an attachment) in case there is _something_ useful here.  In my humble opinion I think the guidance for this should just be "don't move `Ghostty.app` while it is running" 😄 

_Edit: Decided to just send the report to Ghostty for sake of not leaving my full stack out on the internet._

**Comment by pluiedev** [comment, marked as answer]
Not a Mac user, but I don't think moving the binary and resources of any running application on any system is a good idea... Were I to hazard a guess I would say that when you move the executable bundle, Ghostty's resource directory (GHOSTTY_RESOURCES_DIR) will suddenly become nonexistent and any attempt at accessing it would result in crash.
